otx/algorithms/detection/tasks/nncf.py

```diff
@@ -248,46 +226,157 @@ def _create_model(config: Config, from_scratch: bool = False):
             model_cfg.pretrained = None
             logger.warning("build detector")
             model = build_detector(model_cfg)
+
             # Load all weights.
             logger.warning("load checkpoint")
             load_checkpoint(model, init_from, map_location="cpu")
         else:
             logger.warning("build detector")
             model = build_detector(model_cfg)
+
         return model
 
-    def _create_compressed_model(self, dataset, config):
-        init_dataloader = build_dataloader(
-            dataset,
-            config.data.samples_per_gpu,
-            config.data.workers_per_gpu,
-            len(config.gpu_ids),
-            dist=False,
-            seed=config.seed,
-        )
+    def _create_compressed_model(self, config):
+        init_dataloader = build_dataloader(config, "train", False)
+
+        # This data and state dict will be used to build NNCF graph later
+        # when loading NNCF model
+        # because some models run their subcomponents based on intermediate outputs
+        # resulting not a fully traced NNCF graph if fake data is given
+        datasets = get_configs_by_keys(config.data.train, "otx_dataset")
+        for dataset in datasets:
+            self._nncf_data_to_build = dataset[0].numpy
+        self._nncf_state_dict_to_build = copy.deepcopy(self._model.state_dict())
+
         is_acc_aware_training_set = is_accuracy_aware_training_set(config.get("nncf_config"))
 
+        val_dataloader = None
         if is_acc_aware_training_set:
-            self._val_dataloader = build_val_dataloader(config, False)
+            val_dataloader = build_dataloader(config, "val", False)
 
         self._compression_ctrl, self._model = wrap_nncf_model(
             self._model,
             config,
-            val_dataloader=self._val_dataloader,
+            val_dataloader=val_dataloader,
             dataloader_for_init=init_dataloader,
-            get_fake_input_func=get_fake_input,
+            get_fake_input_func=partial(get_fake_input, data=self._nncf_data_to_build),
             is_accuracy_aware=is_acc_aware_training_set,
         )
 
+    def _initialize_post_hook(self):
+        super()._initialize_post_hook()
+
+        # Update model_cfg to initialize model in OTX not MPA
+        initial_model_cfg = copy.deepcopy(self._model_cfg)
+        # merge base model config
+        self._model_cfg.merge_from_dict(dict(model=self._recipe_cfg.model))
+        # merge initial model config
+        self._model_cfg.merge_from_dict(initial_model_cfg)
+
+        # Disable task adaptation in NNCF task
+        self._recipe_cfg.pop("task_adapt", None)
+        self._model_cfg.pop("task_adapt", None)
+        if hasattr(self._recipe_cfg.model, "is_task_adapt"):
+            self._recipe_cfg.model.is_task_adapt = False
+        if hasattr(self._model_cfg.model, "is_task_adapt"):
+            self._model_cfg.model.is_task_adapt = False
```

Comment:
Could you let me know the reason?

Reply:
This key [causes](https://github.com/openvinotoolkit/model_preparation_algorithm/blob/develop/mpa/det/stage.py#L58-L59) to run `configure_task` method in MPA which is not needed for NNCF.